pgjdbc/src/test/java/org/postgresql/test/jdbc42/UpdateObject310Test.java

```
@@ -0,0 +1,127 @@
+    );
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    TestUtil.dropTable(con, "table1");
+    TestUtil.closeDB(con);
+  }
+
+  private Object newValue;
+  private String columnName;
+  private String originalValue;
+
+  public UpdateObject310Test(Object newValue, String columnName, String originalValue) {
+    this.newValue = newValue;
+    this.columnName = columnName;
+    this.originalValue = originalValue;
+  }
+
+  /**
+   * Test the behavior of updateObject for timestamp columns.
+   */
+  @Test
+  public void test() throws SQLException {
+    Statement s = con.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE);
+    ResultSet rs = null;
+
+    try {
+      s.executeUpdate("INSERT INTO table1(" + columnName + ") VALUES(" + originalValue + ")");
+      rs = s.executeQuery(TestUtil.selectSQL("table1", "table1_id, " + columnName));
+      rs.next();
+      rs.updateObject(2, newValue);
+      rs.updateRow();
+      rs.close();
+
+      rs = s.executeQuery(TestUtil.selectSQL("table1", columnName));
+      rs.next();
+      assertEquals(newValue, rs.getObject(1, newValue.getClass()));
+    } finally {
+      if (rs != null) {
+        rs.close();
+      }
+      s.close();
+    }
+  }
+
+  /**
+   * Test the behavior of updateObject for timestamp columns.
+   */
+  @Test
+  public void testWrongClass() throws SQLException {
+    Statement s = con.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE);
+    ResultSet rs = null;
+
+    try {
+      s.executeUpdate("INSERT INTO table1(" + columnName + ") VALUES(" + originalValue + ")");
+      rs = s.executeQuery(TestUtil.selectSQL("table1", "table1_id, " + columnName));
+      rs.next();
+      rs.updateObject(2, 23);
+      rs.updateRow();
+      rs.close();
+
+      fail("should have thrown an Exception");
+    } catch (PSQLException e) {
+      assertEquals(PSQLState.DATATYPE_MISMATCH.getState(), e.getSQLState());
```

Please add error message and keep stacktrace.

Current implementation would throw "assertion error" and it would hide the original exception, stacktrace, etc.

---

I added error messages, but I don't understand how you want me to keep the stacktrace, I tried looking at other unit tests in the code base without finding an example.